### PR TITLE
add test case for #4886

### PR DESCRIPTION
**Pull request by ayushbisht2001**
Add test case 
#### References to other Issues or PRs

refer : #4886

#### Release Notes

NO ENTRY

### CODE REVIEW

**Comment by almighty123789** [comment]
If you have trailing whitespace it will show errors. This one will fix unwanted spaces:
``$ ./bin/strip_whitespace <file>``
You can refer to [this](https://github.com/sympy/sympy/wiki/Development-workflow#be-sure-that-all-tests-of-sympy-pass) for more info.

**Comment by ayushbisht2001** [comment]
@oscarbenjamin , Can you please help me, why code-quality checks are failing . How can I resolve this error ??

**Comment by ayushbisht2001** [comment]
> If you have trailing whitespace it will show errors. This one will fix unwanted spaces:
> `$ ./bin/strip_whitespace <file>`
> You can refer to [this](https://github.com/sympy/sympy/wiki/Development-workflow#be-sure-that-all-tests-of-sympy-pass) for more info.

Thanks

> If you have trailing whitespace it will show errors. This one will fix unwanted spaces:
> `$ ./bin/strip_whitespace <file>`
> You can refer to [this](https://github.com/sympy/sympy/wiki/Development-workflow#be-sure-that-all-tests-of-sympy-pass) for more info.

Thanks @almighty123789

**Comment by ayushbisht2001** [comment]
@oscarbenjamin , Can u please review this..

File: sympy/solvers/tests/test_solvers.py [hunk 1]
```diff
@@ -2314,3 +2314,16 @@ def test_issue_20902():
     assert solve(f.subs({t: 3 * x + 3}).diff(x) > 0, x) == (S(-4)/3 < x) & (x < S(-2)/3)
     assert solve(f.subs({t: 3 * x + 4}).diff(x) > 0, x) == (S(-5)/3 < x) & (x < S(-1))
     assert solve(f.subs({t: 3 * x + 2}).diff(x) > 0, x) == (S(-1) < x) & (x < S(-1)/3)
+
+def test_issue_4886():
+    ra = Symbol('ra', positive=True)
```
Comment (oscarbenjamin) [inline sympy/solvers/tests/test_solvers.py:2319]: There should be two blank lines between `test_` functions. Can you also fix this for the functions immediately above?

File: sympy/solvers/tests/test_solvers.py [hunk 2]
```diff
@@ -2314,3 +2314,16 @@ def test_issue_20902():
...
+    rb = Symbol('rb', positive=True)
+    ra = R**2*a**2
+    rb = R**2*b**2
```
Comment (oscarbenjamin) [inline sympy/solvers/tests/test_solvers.py:2322]: We don't need to set `ra` to a symbol if we are just going to override that on the line below.

Comment (ayushbisht2001) [reply sympy/solvers/tests/test_solvers.py:2321]: @oscarbenjamin  Same for c1, c2, c3, ... ??

**Comment by ayushbisht2001** [comment]
@oscarbenjamin , Now is it good to go ??

File: sympy/solvers/tests/test_solvers.py [hunk 3]
```diff
@@ -2314,3 +2314,14 @@ def test_issue_20902():
...
+
+def test_issue_4886():
+    ra = R**2*a**2
+    rb = R**2*b**2
+    m = a**2 + b**2
+    c1 = b*(((-a*sqrt(ra + rb - c ** 2))) / m + (b * c) / m) - c
+    c2 = ((-a*sqrt(ra + rb - c ** 2)) / m + (b * c) / m)
+    c3 = b*(((a*sqrt(ra + rb - c ** 2))) / m + (b * c) / m) - c
+    c4 = ((a*sqrt(ra + rb - c ** 2)) / m + (b * c) / m)
+    assert solve([x**2 + y**2 - R**2, a*x + b*y - c], x, y) == [(-(c1 / a), c2), (-(c3 / a), c4)]
```
Comment (oscarbenjamin) [inline sympy/solvers/tests/test_solvers.py:2327]: A simpler way to write all of this is:
```python
z = a*sqrt(R**2*a**2 + R**2*b**2 - c**2)/(a**2 + b**2)
t = b*c/(a**2 + b**2)
sol = [((b*(t - z) - c)/(-a), t - z), ((b*(t + z) - c)/(-a), t + z)]
assert solve(...) == sol
```

Comment (ayushbisht2001) [reply sympy/solvers/tests/test_solvers.py:2327]: Okh , I will change it..

**Comment by oscarbenjamin** [comment]
The commits should be squashed. There should be only one commit for a change like this.

**Comment by ayushbisht2001** [comment]
@oscarbenjamin , I don't know why my PR always fails for not having a new line at the end. However I managed to add it correctly, withal it shows me the same error. It only works when I changed the code from github.com.  How can I get rid off this issue??

**Comment by ayushbisht2001** [comment]
@oscarbenjamin ,  would it be eligible for merging ? I was unable to squash the signing and log  commits together to make it a single commit. I didn't find a way to do this.... can u please help me with this...

**Comment by oscarbenjamin** [comment]
It looks like you've squashed most of the commits. There are only two here now.

You can squash those into one with:
```
git checkout issue-4886
git reset HEAD^^
git add sympy/solvers/tests/test_solvers.py
git commit -m "test(solvers): add regression test for GH-4886"
git push --force
```

**Comment by ayushbisht2001** [comment]
thanks oscarbenjamin...

**Comment by ayushbisht2001** [comment]
@oscarbenjamin , is it better now ??

**Comment by ayushbisht2001** [comment]
whenever you get some free time, can u please review this #21097,  one also .

**Comment by oscarbenjamin** [comment]
This looks good. Thanks.

**Pull request merged**